### PR TITLE
Fix deserialize-custom-data-frontend

### PR DESCRIPTION
**Pull request by EmekaManuel**
fix: replace deprecated findProgramAddress with findProgramAddressSync
refactor: improve error handling and logging in Movie deserialization

### Problem
- The method 'web3.PublicKey.findProgramAddress' is deprecated and may lead to compatibility issues.
- Insufficient error handling and logging during movie deserialization made debugging difficult.

### Summary of Changes
- Replaced the deprecated `findProgramAddress` method with the recommended `findProgramAddressSync` method for more reliable and up-to-date usage.
- Added detailed logging for buffer length and raw buffer data in hexadecimal format to aid in troubleshooting.
- Switched from `console.log` to `console.error` for better clarity and appropriate error handling.

### CODE REVIEW

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Same feedback as https://github.com/solana-foundation/developer-content/pull/346, can you also please ensure the screenshots are updated to show your new program?

You will also need to update https://github.com/Unboxed-Software repos mentioned in this article (or if you have replaced these repos, update the URLs). This is mentioned on the Superteam Earn bounty.

One these are fixed this should be a good candidate to go in.

**Comment by EmekaManuel** [comment]
i have made the necessary changes which include: 
- Replaced abbreviations like e with error  with more descriptive variable names for better code readability.
- Improved efficiency by importing specific modules from @solana/web3.js instead of importing the entire package.
- Applied linting to maintain code consistency and clarity.

**Comment by EmekaManuel** [comment]
once i rebuild the new application using create-solana-dapp, I'd update the unboxed repo links and include screenshots.

File: content/courses/native-onchain-development/deserialize-custom-data-frontend.md [hunk 1]
```diff
@@ -273,11 +273,14 @@ export class Movie {
...
-      console.log('Deserialization error:', error)
-      return null
+      const { title, rating, description } =
+        this.borshAccountSchema.decode(buffer);
+      return new Movie(title, rating, description);
+    } catch (e) {
+      console.error("Deserialization error:", e);
+      console.error("Buffer length:", buffer.length);
+      console.error("Buffer data:", buffer.toString("hex"));
+      return null;
```
[the remark this answers is not in the document]
Comment (EmekaManuel) [reply content/courses/native-onchain-development/deserialize-custom-data-frontend.md:284]: fixed.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Waiting for create-solana-dapp and screenshots as previously mentioned.

**Comment by EmekaManuel** [comment]
> Waiting for create-solana-dapp and screenshots as previously mentioned.

I have done these : 
Screenshot 2024-09-06 at 09 19 24

**Review by mikemaccana — APPROVED** [review]
Thanks @EmekaManuel ! Marking this as the winner for deserialize-custom-data-frontend. 🏆

**Pull request merged**